### PR TITLE
Update disk format related function.

### PR DESCRIPTION
**Pull request by XueqiangWei**
ID: 1422740  1561839
Signed-off-by: Xueqiang Wei <xuwei@redhat.com>

### CODE REVIEW

File: virttest/utils_misc.py [hunk 1]
```diff
@@ -2834,96 +2846,120 @@ def get_all_disks_did(session, partition=False):
...
+        if not mountpoint:
+            session.cmd(format_cmd % (did, fstype), timeout=300)
+            details = session.cmd_output(detail_cmd % did)
+            for line in details.splitlines():
+                pattern = "\s+Volume\s+\d+"
+                if re.search(pattern, line, re.I | re.M):
+                    mountpoint_list.append(line.split()[2])
+        else:
+            session.cmd(format_cmd % (did, mountpoint[i], fstype), timeout=300)
+    return mountpoint or mountpoint_list
```
Comment (aliang123) [inline virttest/utils_misc.py]: "return a list: mount point list for all partitions" in the doc, but
"return mountpoint or mountpoint_list" -- a boolean value in function.

Comment (XueqiangWei) [reply virttest/utils_misc.py]: it is a list.
e.g.
>>> [ ] or [\"a\", \"b\"]
['a', 'b']
>>> [\"a\", \"b\"] or [ ]
['a', 'b']
>>>

File: virttest/utils_misc.py [hunk 2]
```diff
@@ -2933,71 +2969,117 @@ def format_linux_disk(session, did, all_disks_did, partition=False,
...
+            logging.error("Invalid size value: '%s'" % size)
+            return []
+        devname = '/dev/%s' % kname
+        partition_list = []
+        mountpoint_list = []
+        if partition:
+            partition_list.append(kname)
+        else:
+            logging.info("Create partition on disk %s" % devname)
+            partition_size = int(size[0:-1])/n_partitions
```
Comment (lolyu) [inline virttest/utils_misc.py]: partition_size should be a float, if size is 1G and n_partitions is 2, partition_size will get 0.

Comment (XueqiangWei) [reply virttest/utils_misc.py]: updated, thanks

**Comment by XueqiangWei** [comment]
Need to rebase it

**Comment by XueqiangWei** [comment]
rebase & resend this pullreq, so close it.

**Pull request closed without merging**